### PR TITLE
I2C hot switch master <-> slave

### PR DESCRIPTION
**Pull request by maidnl**
Solve: https://github.com/arduino/ArduinoCore-renesas/issues/180
This PR introduces the possibility to change back and forth the role on the I2C from Master to Slave and vice versa.

### CODE REVIEW

File: libraries/Wire/Wire.cpp [hunk 1]
```diff
@@ -252,8 +253,9 @@ bool TwoWire::cfg_pins(int max_index) {
     ioport_sda = USE_SCI_EVEN_CFG(cfg_sda) ? IOPORT_PERIPHERAL_SCI0_2_4_6_8 : IOPORT_PERIPHERAL_SCI1_3_5_7_9;
     ioport_scl = USE_SCI_EVEN_CFG(cfg_scl) ? IOPORT_PERIPHERAL_SCI0_2_4_6_8 : IOPORT_PERIPHERAL_SCI1_3_5_7_9;
   
-    R_IOPORT_PinCfg(&g_ioport_ctrl, g_pin_cfg[sda_pin].pin, (uint32_t) (IOPORT_CFG_PULLUP_ENABLE | IOPORT_CFG_PERIPHERAL_PIN | ioport_sda));
-    R_IOPORT_PinCfg(&g_ioport_ctrl, g_pin_cfg[scl_pin].pin, (uint32_t) (IOPORT_CFG_PULLUP_ENABLE | IOPORT_CFG_PERIPHERAL_PIN | ioport_scl));
+    fsp_err_t err = R_IOPORT_PinCfg(&g_ioport_ctrl, g_pin_cfg[sda_pin].pin, (uint32_t) (IOPORT_CFG_PULLUP_ENABLE | IOPORT_CFG_PERIPHERAL_PIN | ioport_sda));
```
Comment (aentinger) [inline libraries/Wire/Wire.cpp:256]: Is `err` used anywhere? Otherwise it can probably be dropped. Otherwise this looks like nothing else has changed.

**Review by aentinger — CHANGES_REQUESTED** [review]
LGTM :+1: I believe `err` can be dropped. Let's do that and than this PR can be merged :+1:

**Comment by facchinm** [comment]
Before merging I'd try a cleanup to get the minimum amount of changes in IRQManager (I know it might be hard but it could avoid cross references difficult to spot and fix afterwards)

File: libraries/Wire/Wire.cpp (hunk 1 above)
Comment (maidnl) [reply libraries/Wire/Wire.cpp:256]: @aentinger that was a "left over" from my debug. Removed.

Comment (aentinger) [reply libraries/Wire/Wire.cpp:256]: No problem :wink: thanks for the update :+1:

**Comment by maidnl** [comment]
This PR is not longer needed because this new PR https://github.com/arduino/ArduinoCore-renesas/pull/191 supersedes this one

**Pull request closed without merging**